code/espurna/thinkspeak.ino

```diff
@@ -96,6 +102,11 @@ void _tspkConfigure() {
         setSetting("tspkEnabled", 0);
     }
     if (_tspk_enabled && !_tspk_client) _tspkInitClient();
+
+    URL _url(getSetting("tspkAddress", THINGSPEAK_ADDRESS));
+    _tspk_host = _url.host;
+    _tspk_url = _url.path;
+    _tspk_port = _url.port;
```

Comment:
Nothing changed though? 

Just to clarify:
- SDK and network run in a separate light "thread"
- loop() / setup() run in a different thread, which can be paused at any time via `delay(...)` or `yield()`
- configuration callbacks run at the start of the `loop()`

Sync methods of thingspeak don't need a global URL to track things, since they start the network request and exit in the same function. Async schedules next `yield()` / `delay()` to execute network request in a system thread, thus we need a persistent variable to store host/port/path tuple. If we make a request and change configuration via configure callback, by the time system reaches network callbacks we can have a different value.

Reply:
I understand that it is not atomic, but I use this URL for notifications to a simple PHP script. So if I lose one or more data, it doesn't matter for me. Moreover, I don't see in which case it could be annoying. However, I'm ready to make the changes you want, but I need you to be more explicit: tell me what you want me to change.

Reply:
What I was thinking is:
- parse URL in tspkFlush(), pass as argument to Post()
- sync function references arg, async copies to a global and uses that

Reply:
Hi Max, I pushed a new version. In this one, I did a AsyncThingspeak object that inherit from AsyncClient and contain the URL.

I must admit that c ++ is not my usual language, and I'm not as comfortable as in other languages. So, if you have advices to give me about this code, I am listening to all proposals.

However, I need help with the THINGSPEAK_USE_ASYNC directive: the _tspkConfigure() function which is in the general context, calls the _tspkInitClient() function which only exists in the THINGSPEAK_USE_ASYNC context. Is this a bug, or did I miss something ?

Reply:
Yeah, sync client isn't specifically tested all the time, that is an oversight. `#if ...` preprocessor block can strip that code part.